htdocs/societe/list.php

```diff
@@ -423,9 +449,33 @@
 $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_departements as state on (state.rowid = s.fk_departement)";
 $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_regions as region on (region.	code_region = state.fk_region)";
 // We'll need this table joined to the select in order to filter by categ
-if (!empty($search_categ_cus)) $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX."categorie_societe as cc ON s.rowid = cc.fk_soc"; // We'll need this table joined to the select in order to filter by categ
+if (!empty($searchCategoryCustomerList)) {
+	if ($searchCategoryCustomerOperator == 1) {
```

Comment:
To manage the AND, why not just add a criteria :
AND (select count(*) from categorie_societe where id in (A, B, C) where fk_soc =s.fk_soc) = 3
We will soon introduce an advanced filter selector to allow to combine any field with or and and and we must have simple rules to add such test that are compatible without changing the request.

Reply:
It's a good idea, I will try to replace this part of code